core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -484,20 +484,36 @@ default Executor makeContextAware(Executor executor) {
 
     /**
      * Returns an {@link ExecutorService} that will execute callbacks in the given {@code executor}, making
-     * sure to propagate the current {@link RequestContext} into the callback execution.
+     * sure to propagate this {@link RequestContext} into the callback execution.
      */
     default ExecutorService makeContextAware(ExecutorService executor) {
         return ContextAwareExecutorService.of(this, executor);
     }
 
+    /**
+     * Returns an {@link ExecutorService} that will execute callbacks in the given {@code executor}, propagating
+     * the caller's {@link RequestContext} (if any) into the callback execution.
+     */
+    static ExecutorService makeContextPropagating(ExecutorService executor) {
+        return new PropagatingContextAwareExecutorService(executor);
+    }
```

Comment:
I wonder if `RequestContext` is a good place to put these new methods given this interface is quite crowded already. Should we introduce a new utility class, like `ContextAwareExecutors`, e.g. `ContextAwareExecutors.makeContextPropagating()`?

Reply:
I think it will be very confusing if the new methods and the old ones aren't in the same place, and presumably it's not possible to move the existing methods.